src/els_indexer.erl

```diff
@@ -211,3 +212,28 @@ register_reference(Uri, #{id := {F, A}, range := Range}) ->
 add_reference(Key, Value) ->
   UpdateFun = fun(OldRefs) -> ordsets:add_element(Value, OldRefs) end,
   ok = els_db:update(references, Key, UpdateFun, ordsets:new()).
+
+%% @edoc Remove all references to a given uri()
+-spec purge_reference_uri(uri()) -> ok.
```

Comment:
We are removing every object and reinserting all of the ones that are not pointing to the URI. This is a lot of work to do every time a file is reindexed (which currently happens on every file change).

An alternative would be to change the way we store the references. Instead of having a table of type `set` we could change `references` to be a `bag` or `duplicate_bag` table and then use something like `ets:select_delete/2` to remove all objects where the `uri` is the one we want to remove.

```
MatchSpec = ets:fun2ms(fun({K, #{uri := U}}) -> U =:= Uri end),
_DeletedCount = ets:select_delete(references, MatchSpec),
ok.
```

Changing the `references` table will require changes in the indexing here and in the `els_references_provider`.

WDYT?

Reply:
I agree.  I see this current PR as a draft to expose the requirements and get some feedback.